internal/events/stream/event.go

```diff
@@ -0,0 +1,52 @@
+package stream
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/openshift/assisted-service/internal/versions"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/stream"
+)
+
+type EventEnvelope struct {
+	Name     string        `json:"name"`
+	Payload  interface{}   `json:"payload"`
+	Metadata EventMetadata `json:"metadata"`
+}
+
+type EventMetadata struct {
+	Versions *models.Versions `json:"versions"`
+}
+
+// utility function to stream event to event stream
+func StreamEvent(ctx context.Context, stream stream.EventStreamWriter, eventName, key string, payload interface{}) error {
+	if stream == nil {
+		return errors.New("Event stream not configured")
+	}
+	envelope := &EventEnvelope{
+		Name:    eventName,
+		Payload: payload,
+	}
+	v, err := getVersions()
+	if err == nil {
+		envelope.Metadata.Versions = v
+	}
+	streamPayload, err := json.Marshal(envelope)
+	if err != nil {
+		return err
+	}
+	return stream.Write(ctx, []byte(key), streamPayload)
+}
+
+func getVersions() (*models.Versions, error) {
+	v := versions.Versions{}
+	err := envconfig.Process("", &v)
```

Comment:
Good point, as discussed somewhere else it would be best to make it more generic and inject it in the entrypoint instead - thus freeing this package from the `internal` bit.